l1-contracts/test/foundry/unit/concrete/Bridges/L1SharedBridge/L1SharedBridgeFails.t.sol

```diff
@@ -14,9 +14,165 @@ import {IMailbox} from "contracts/state-transition/chain-interfaces/IMailbox.sol
+
+        vm.expectRevert("ShB: legacy bridge already set");
+        sharedBridge.setL1Erc20Bridge(anotherBridge);
+        vm.stopPrank();
+    }
+
+    function test_bridgehubDepositBaseToken_callerNotBridgeHubOrEra(address caller, uint256 chainId) public {
+        vm.assume(caller != bridgehubAddress);
+        vm.assume(chainId != chainId || caller != eraDiamondProxy);
+
+        vm.deal(caller, amount);
+        vm.expectRevert("L1SharedBridge: not bridgehub or era chain");
+        vm.prank(caller);
+        sharedBridge.bridgehubDepositBaseToken{value: amount}(chainId, alice, ETH_TOKEN_ADDRESS, amount);
+    }
+
+    function test_receiveEth_notSTM(uint256 amount, address caller) public {
+        address stmAddress = makeAddr("stm");
+
+        vm.mockCall(
+            bridgehubAddress,
+            abi.encodeWithSelector(IBridgehub.getHyperchain.selector, eraChainId),
+            abi.encode(stmAddress)
+        );
+
+        vm.assume(caller != stmAddress);
+        vm.deal(caller, amount);
+        vm.prank(caller);
+        vm.expectRevert("receiveEth not state transition");
+        sharedBridge.receiveEth{value: amount}(eraChainId);
+    }
+
+    function test_setEraPostDiamondUpgradeFirstBatch_wrongValue(uint256 eraPostUpgradeFirstBatch) public {
+        eraPostUpgradeFirstBatch = bound(eraPostUpgradeFirstBatch, 1, type(uint256).max);
+
+        vm.prank(owner);
+        sharedBridge.setEraPostDiamondUpgradeFirstBatch(eraPostUpgradeFirstBatch);
+
+        vm.prank(owner);
+        vm.expectRevert("ShB: eFPUB already set");
+        sharedBridge.setEraPostDiamondUpgradeFirstBatch(eraPostUpgradeFirstBatch);
+    }
+
+    function setEraPostLegacyBridgeUpgradeFirstBatch_wrongValue(uint256 eraPostLegacyBridgeUpgradeFirstBatch) public {
+        eraPostUpgradeFirstBatch = bound(eraPostLegacyBridgeUpgradeFirstBatch, 1, type(uint256).max);
+
+        vm.prank(owner);
+        sharedBridge.setEraPostLegacyBridgeUpgradeFirstBatch(eraPostLegacyBridgeUpgradeFirstBatch);
+
+        vm.prank(owner);
+        vm.expectRevert("ShB: eFPUB already set");
+        sharedBridge.setEraPostLegacyBridgeUpgradeFirstBatch(eraPostLegacyBridgeUpgradeFirstBatch);
+    }
+
+    function test_setEraLegacyBridgeLastDepositTime_batchAlreadySet(
+        uint256 eraLegacyBridgeLastDepositBatch,
+        uint256 eraLegacyBridgeLastDepositTxNumber
+    ) public {
+        L1SharedBridge sharedBridgeImpl = new L1SharedBridge({
+            _l1WethAddress: l1WethAddress,
+            _bridgehub: IBridgehub(bridgehubAddress),
+            _eraChainId: eraChainId,
+            _eraDiamondProxy: eraDiamondProxy
+        });
+        TransparentUpgradeableProxy sharedBridgeProxy = new TransparentUpgradeableProxy(
+            address(sharedBridgeImpl),
+            admin,
+            abi.encodeWithSelector(L1SharedBridge.initialize.selector, owner)
+        );
+
+        L1SharedBridge sharedBridge = L1SharedBridge(payable(sharedBridgeProxy));
```

Comment:
Is it necessary to create new SharedBridge?

Reply:
it's needed to check if setter set the values, another approach would be to remove setters from constructor and use it in any method needed

Reply:
Address of legacy bridge seems like something that not changes for me, other setters the other way around.

Reply:
Proxy was sometimes throwing errors, I've replaced it with ownership transfer.